server/pbenchinacan/container-build.sh

```diff
@@ -79,5 +81,31 @@ buildah run $container systemctl enable nginx
 buildah run $container systemctl enable rsyslog
 buildah run $container systemctl enable pbench-server
 
-# Create the container image.
-buildah commit $container localhost/pbench-server:${PB_SERVER_IMAGE_TAG}
+# Skip installing and configuring the Firewall
+
+# Copy the Pbench Server config file for our "in-a-can" environment, and
```

Comment:
I am not in favor of baking these steps (lines 86 thru 108) into the image.  I believe they should be separate steps that are part of the deployment specific to the CI environment.

Reply:
I think it's a trade-off.

I believe that doing it this way has only a tiny effect on the Staging environment, but I think it has a significant benefit for the other two scenarios which this image is intended to support, and so it's worth using this approach.

For the Staging environment, all that this does is make the container image slightly larger.  These files don't actually impact the Staging deployment, because they are overmapped with local volumes when the container is run.

For the other environments, it means that these files and directories don't have to be created on the fly, they don't have to be allocated somewhere in the local environment and mapped in, and they don't have to be cleaned up after the container exits.  I think the simplicity of that management is worth the relatively tiny cost to the Staging case.

Reply:
The principle that we are trying to follow is that our functional test environment does not deviate from what will be done with staging or production.

In this case, have an image we deploy in production where a /srv/pbench volume exists internal to the image means that a mistake in the production deployment mechanism does not result in a missing /srv/pbench volume but one where the server will start up using the internal /srv/pbench and cause us more headaches debugging.

Reply:
> means that a mistake in the production deployment mechanism does not result in a missing /srv/pbench volume

This is true.  And, I concur that there is value in that.

Unfortunately, going that way makes standing up functional testing more complicated, which carries its own costs in time and effort.

But, if that's what you want, I'll move that stuff to somewhere else.

Reply:
I hope we move towards adopting the principle because we want to be operationally focused and not development focused.

Added time and effort incurred supporting the principle is well worth it compared to problem encountered in production.

Reply:
I think it would be best to Nikhil's changes to `server/pbenchinacan/run-pbench-in-a-can` in place in order to have a reasonable home for doing the local Pbench Server deployment, so it would be good if we could streamline moving #3216 forward.

Reply:
Per our discussion, let's merge this PR and make this change on top o f #3216 once it lands.  Thanks, @portante!